packages/fxa-settings/src/lib/model-data/data-stores/location-state-data.ts

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { ReachRouterWindow } from '../../window';
+import { ModelDataStore } from '../model-data-store';
+
+/**
+ * A data store that wraps Reach Router's `window.location.state`.
+ */
+export class LocationStateData extends ModelDataStore {
+  constructor(protected readonly window: ReachRouterWindow) {
+    super();
+
+    if (!window.location.state) {
+      window.location.state = {};
```

Comment:
If I'm reading this correctly, this is mutating the state of something a couple levels deep in the argument.  I guess presumably that value isn't also directly used by some other code elsewhere (or it doesn't matter) so we can mess with it?

Reply:
This mutation only ensures that state exists, and I think that particular modificatonis safe. We woudln't want to modified the state of the object itself because those values might be referenced else where. From what I've see, this state object is fairly transient, and just used to pass values between pages. I'll change this to if window.location.state == null to make the intent more clear.